### DISCUSSION TITLE
Reasonable to return NO_ADDITIONAL_SAS on child SA explosions?

### DISCUSSION THREAD

**Discussion by jamesyonan** [strongswan/strongswan#2563] [General]
We are trying to deal with some IPSec clients (out of our control) that generate numerous child SAs without limit.  In `build_r()` in *child_create.c*, is it reasonable to check the current number of SAs via `ike_sa->get_child_count()` and return `NO_ADDITIONAL_SAS` if the maximum count is exceeded?  See attached patch.

Thanks,
James

[max_child.PATCH](https://github.com/user-attachments/files/18088099/max_child.PATCH)

**Comment by tobiasbrunner** [comment]
If the peer understands it that way, maybe (strongSwan, for instance, doesn't). And it really depends on whether there are different CHILD_SAs (i.e. with different traffic selectors) under the IKE_SA or not, as the notify sets a hard stop on all of them, not a specific one. Note that the notify is also used to reject rekeyings.
